shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -56,51 +91,40 @@ public void acquireLatestImage() {
   @Override
   protected void onDraw(Canvas canvas) {
     super.onDraw(canvas);
-    if (nextImage == null) {
-      return;
-    }
-
-    if (currentImage != null) {
-      currentImage.close();
+    if (nextImage != null) {
+      if (currentImage != null) {
+        currentImage.close();
+      }
+      currentImage = nextImage;
+      nextImage = null;
+      updateCurrentBitmap();
     }
-    currentImage = nextImage;
-    nextImage = null;
 
-    if (android.os.Build.VERSION.SDK_INT >= 29) {
-      drawImageBuffer(canvas);
-      return;
+    if (currentBitmap != null) {
+      canvas.drawBitmap(currentBitmap, 0, 0, null);
     }
-
-    drawImagePlane(canvas);
   }
 
   @TargetApi(29)
-  private void drawImageBuffer(@NonNull Canvas canvas) {
-    final HardwareBuffer buffer = currentImage.getHardwareBuffer();
-
-    final Bitmap bitmap = Bitmap.wrapHardwareBuffer(buffer, ColorSpace.get(ColorSpace.Named.SRGB));
-    canvas.drawBitmap(bitmap, 0, 0, null);
-  }
-
-  private void drawImagePlane(@NonNull Canvas canvas) {
-    if (currentImage == null) {
-      return;
-    }
-
-    final Plane[] imagePlanes = currentImage.getPlanes();
-    if (imagePlanes.length != 1) {
-      return;
+  private void updateCurrentBitmap() {
+    if (android.os.Build.VERSION.SDK_INT >= 29) {
+      final HardwareBuffer buffer = currentImage.getHardwareBuffer();
+      currentBitmap = Bitmap.wrapHardwareBuffer(buffer, ColorSpace.get(ColorSpace.Named.SRGB));
+    } else {
+      final Plane[] imagePlanes = currentImage.getPlanes();
+      if (imagePlanes.length != 1) {
+        return;
+      }
+
+      final Plane imagePlane = imagePlanes[0];
+      final int desiredWidth = imagePlane.getRowStride() / imagePlane.getPixelStride();
+      final int desiredHeight = currentImage.getHeight();
+
+      currentBitmap =
+          android.graphics.Bitmap.createBitmap(
```

Comment:
nit: `Bitmap.createBitmap`

Feel free to address this later: The Android team suggested to reuse the bitmap.

Reply:
done